packages/main-api/src/utils/aws_chime_sdk_meeting.rs

```diff
@@ -241,58 +242,143 @@ impl ChimeMeetingService {
 
         Ok(())
     }
+
+    pub async fn move_meeting_artifacts_with_retry(
+        &self,
+        bucket_name: &str,
+        meeting_id: &str,
+    ) -> Result<()> {
+        let prefix = format!("{}/", meeting_id);
+        let mut attempt = 0;
+        let max_attempts = 10;
+        let delay = Duration::from_secs(2);
+
+        loop {
+            attempt += 1;
+
+            let list_result = self
+                .s3
+                .list_objects_v2()
+                .bucket(bucket_name)
+                .prefix(&prefix)
+                .send()
+                .await
+                .map_err(|e| {
+                    tracing::error!("S3 list_objects_v2 failed on attempt {}: {:?}", attempt, e);
+                    Error::AwsS3Error(e.to_string())
+                })?;
+
+            if let Some(objects) = list_result.contents {
+                if !objects.is_empty() {
+                    tracing::info!("S3 artifacts found after {} attempt(s)", attempt);
+                    return self.move_meeting_artifacts(bucket_name, meeting_id).await;
+                }
+            }
+
+            if attempt >= max_attempts {
+                tracing::warn!(
+                    "No artifacts found after {} attempts (~{}s). Skipping move.",
+                    attempt,
+                    attempt * delay.as_secs()
+                );
+                return Ok(());
+            }
+
+            tracing::debug!(
+                "Waiting for artifacts... attempt {}/{}. Retrying in {}s",
+                attempt,
+                max_attempts,
+                delay.as_secs()
+            );
+            sleep(delay).await;
+        }
+    }
+
+    pub async fn move_meeting_artifacts(&self, bucket_name: &str, meeting_id: &str) -> Result<()> {
+        let conf = crate::config::get();
+
+        let list_result = self
+            .s3
+            .list_objects_v2()
+            .bucket(bucket_name)
+            .prefix(format!("{}/", meeting_id))
+            .send()
+            .await
+            .map_err(|e| {
+                tracing::error!("S3 list_objects error: {:?}", e);
+                Error::AwsS3Error(e.to_string())
+            })?;
+
+        if let Some(objects) = list_result.contents {
+            for obj in objects {
+                if let Some(key) = obj.key {
+                    let filename = key.split('/').last().unwrap_or("artifact");
+
+                    let folder = if key.contains("/video/") {
+                        "video"
+                    } else if key.contains("/audio/") {
+                        "audio"
+                    } else if key.contains("/meeting-events/") {
+                        "meeting-events"
+                    } else {
+                        "etc"
+                    };
+
+                    let destination_key =
+                        format!("{}/{}/{}/{}", conf.env, meeting_id, folder, filename);
```

Comment:
ContentType 또는 FolderType enum 을 정의하고요.
FromStr 또는 From<String> 을 트레이트와 Display 트레이트를 구현해서 아래와 같이 되는 형태가 되어야 합니다.

```suggestion
                    let folder:FolderType = key.parse()?; // 또는 let folder:FolderType = key.into();

                    let destination_key =
                        format!("{}/{}/{}/{}", conf.env, meeting_id, folder, filename);
```

Reply:
enum 형태로 수정했습니다.